src/workstation/core.py

```diff
@@ -151,6 +152,14 @@ def create_workstation(
         env["no_proxy"] = no_proxy
         env["NO_PROXY"] = no_proxy
 
+    if envs:
+        user_envs = dict(envs)
+        # ensure that no duplicate keys are added to env
+        for key, value in user_envs.items():
+            if key not in env:
+                env[key] = value
```

Comment:
nit: log the duplicate being ignored or just add validation to click to fail in this scenario